### PR TITLE
feat(ui): add a copyable ip address to pod details

### PR DESCRIPTION
**Pull request by huwshimi**
Add a copyable code component containing a Pod's IP address to the pod details page.

## Done

- Add a copyable code component containing a Pod's IP address to the pod details page.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)

### QA steps

- Navigate to the details page for a pod.
- There should be a code box at the top with the ip address, à la: https://app.zeplin.io/project/5f1a24be5ee8af8723f4e7fa/screen/5f315c75b688551f4783d49b

## Fixes

Fixes: canonical-web-and-design/MAAS-squad#2071.

### CODE REVIEW

File: ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx [hunk 1]
```diff
@@ -27,8 +27,18 @@ const KVMSummary = (): JSX.Element => {
   }, [dispatch]);
 
   if (!!pod) {
+    // Fetch the IP address that is contained in the power_address. This is the
+    // only place the IP address is exposed.
+    const ip_address = pod.power_address.match(/[\d.]+/)[0];
```
Comment (Caleb-Ellis) [inline ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx:32]: Was it decided to only show the ip portion of a power address now? The legacy version just shows the whole power address.

File: ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx [hunk 2]
```diff
@@ -27,8 +27,18 @@ const KVMSummary = (): JSX.Element => {
...
     return (
       <>
+        <div className="u-flex">
+          <p className="u-nudge-left">LXD URL:</p>
```
Comment (Caleb-Ellis) [inline ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx:36]: This text depends on the pod type. In the legacy version, virsh pods = "Virsh:" lxd pods = "LXD URL:"

**Review by Caleb-Ellis — COMMENTED** [review]
Just a question and text change

File: ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx:32]: Ah right, was just going by the doc which has:
> IP address URL: copyable ip address for the instance.

@amylily1011 should this just replicate what already exists? Examples below:

Screen Shot 2020-08-20 at 9 27 00 am
Screen Shot 2020-08-20 at 9 27 47 am

Comment (amylily1011) [reply ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx:32]: Hi @Caleb-Ellis and @huwshimi, we're just going to replicate the legacy version for this one. No new decisions have been made.

Comment (huwshimi) [reply ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.tsx]: Thanks @amylily1011.

@Caleb-Ellis I've updated this now.

File: ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.test.tsx [hunk 3]
```diff
@@ -47,4 +47,29 @@ describe("KVMSummary", () => {
     expect(wrapper.find("KVMAggregateResources").exists()).toBe(false);
     expect(wrapper.find("KVMNumaResources").exists()).toBe(true);
   });
+
+  it("can display the ip address", () => {
```
Comment (Caleb-Ellis) [inline ui/src/app/kvm/views/KVMDetails/KVMSummary/KVMSummary.test.tsx:51]: Maybe this should be "power address"

**Review by Caleb-Ellis — APPROVED** [review]
LGTM!

**Pull request merged**